test/e2e/flowaggregator_test.go

```diff
@@ -427,40 +430,63 @@ func testHelper(t *testing.T, data *TestData, podAIPs, podBIPs, podCIPs, podDIPs
+			checkRecordsForFlows(t, data, podAIPs.ipv6.String(), getIperfServerIP(t, externalServer, true), isIPv6, ipfixregistry.FlowTypeToExternal, false, false, false, false)
+		}
+	})
+
 	// LocalServiceAccess tests the case, where Pod and Service are deployed on the same Node and their flow information is exported as IPFIX flow records.
 	t.Run("LocalServiceAccess", func(t *testing.T) {
 		skipIfProxyDisabled(t, data)
 		if !isIPv6 {
-			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcB.Spec.ClusterIP, isIPv6, true, true, false, false, checkBandwidth)
+			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcB.Spec.ClusterIP, isIPv6, ipfixregistry.FlowTypeIntraNode, true, false, false, checkBandwidth)
 		} else {
-			checkRecordsForFlows(t, data, podAIPs.ipv6.String(), svcB.Spec.ClusterIP, isIPv6, true, true, false, false, checkBandwidth)
+			checkRecordsForFlows(t, data, podAIPs.ipv6.String(), svcB.Spec.ClusterIP, isIPv6, ipfixregistry.FlowTypeIntraNode, true, false, false, checkBandwidth)
 		}
 	})
 
 	// RemoteServiceAccess tests the case, where Pod and Service are deployed on different Nodes and their flow information is exported as IPFIX flow records.
 	t.Run("RemoteServiceAccess", func(t *testing.T) {
 		skipIfProxyDisabled(t, data)
 		if !isIPv6 {
-			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcC.Spec.ClusterIP, isIPv6, false, true, false, false, checkBandwidth)
+			checkRecordsForFlows(t, data, podAIPs.ipv4.String(), svcC.Spec.ClusterIP, isIPv6, ipfixregistry.FlowTypeInterNode, true, false, false, checkBandwidth)
 		} else {
-			checkRecordsForFlows(t, data, podAIPs.ipv6.String(), svcC.Spec.ClusterIP, isIPv6, false, true, false, false, checkBandwidth)
+			checkRecordsForFlows(t, data, podAIPs.ipv6.String(), svcC.Spec.ClusterIP, isIPv6, ipfixregistry.FlowTypeInterNode, true, false, false, checkBandwidth)
 		}
 	})
 }
 
-func checkRecordsForFlows(t *testing.T, data *TestData, srcIP string, dstIP string, isIPv6 bool, isIntraNode bool, checkService bool, checkK8sNetworkPolicy bool, checkAntreaNetworkPolicy bool, checkBandwidth bool) {
+func checkRecordsForFlows(t *testing.T, data *TestData, srcIP string, dstIP string, isIPv6 bool, flowType uint8, checkService bool, checkK8sNetworkPolicy bool, checkAntreaNetworkPolicy bool, checkBandwidth bool) {
 	timeStart := time.Now()
 	timeStartSec := timeStart.Unix()
-	var cmdStr string
-	if !isIPv6 {
-		cmdStr = fmt.Sprintf("iperf3 -c %s -t %d|grep sender|awk '{print $7,$8}'", dstIP, iperfTimeSec)
-	} else {
-		cmdStr = fmt.Sprintf("iperf3 -6 -c %s -t %d|grep sender|awk '{print $7,$8}'", dstIP, iperfTimeSec)
+
+	// specific to checkToExternalFlows test: doing a for loop here since the connection to iperf server is unstable,
+	// we will try to re-connect to server if stdout is empty, showing srcPod not connected to server in the last try
+	var cmdStr, stdout string
+	var err error
+	var duration time.Duration
+	for stdout == "" && duration.Seconds() < collectorCheckTimeout.Seconds() {
```

Comment:
Use wait.Poll instead of for loop?

Reply:
Thanks! 
1. iperf pod cannot connect to google.com by doing `iperf3 -c dstIP`, also we cannot do `ping google` since there's no ping installed on the container. It can only connect to iperf server or other iperf pod serving as a server.
2. Actually I didn't do the bandwidth check by setting the checkBandwidth to false at [L437](https://github.com/antrea-io/antrea/blob/030bfc13b131b17e7ab7d8c678a55a194555ccfe/test/e2e/flowaggregator_test.go#L437). Previously I didn't separate the pod-to-external flow because I wanted to re-use the codes making the iperf connection. Now I separate them by flow type to make it clear.
3. Changed to wait.Poll.

Reply:
We should not use iperf at all I think for the Pod-to-external test. Maybe just an HTTP get request with curl or wget.

Reply:
Agree. Re-wrote the test by using a busybox pod and ping google.